scripts/globals/events/egg_hunt_egg-stravaganza.lua

```diff
@@ -0,0 +1,898 @@
+            local secondLetter = string.byte(string.lower(second)) - 97
+
+            if
+                firstLetter == (option - 3) and
+                party[i]:getEquipID(xi.slot.HEAD) == xi.items.EGG_HELM
+            then
+                return xi.items.A_EGG + secondLetter
+            end
+        end
+    end
+
+    return 0
+end
+
+xi.events.egg_hunt.combos =
+{
+    { check = firstThree,    message = messageOffset.REWARD1 },
+    { check = sevenKind,     message = messageOffset.REWARD2 },
+    { check = straightEight, message = messageOffset.REWARD2 },
+    { check = regionControl, message = messageOffset.REWARD2 },
+    { check = weekDay,       message = messageOffset.REWARD2 },
+}
+
+if settings.ERA_2014 then
+    table.insert(xi.events.egg_hunt.combos, { check = era2014, message = messageOffset.REWARD2 })
+end
+
+if settings.ERA_2015 then
+    table.insert(xi.events.egg_hunt.combos, { check = era2015, message = messageOffset.REWARD2 })
+end
+
+if settings.ERA_2018 then
+    table.insert(xi.events.egg_hunt.combos, { check = era2018, message = messageOffset.REWARD2 })
+end
+
+for k, v in pairs(settings.BONUS_WORDS) do
+    local customEggs = xi.events.egg_hunt.stringToEggs(k)
+
+    table.insert(xi.events.egg_hunt.combos,
+    {
+        check = function(player, npc, trade)
+            if npcUtil.tradeHasExactly(trade, customEggs) then
+                return v
+            end
+        end,
+
+        message = messageOffset.REWARD2,
+    })
+end
+
+------------------------------------
+-- Moogle event handlers
+------------------------------------
+
+xi.events.egg_hunt.onTrigger = function(player, npc)
+    local zoneID = player:getZoneID()
+
+    if player:getCharVar(settings.VAR.DAILY_EGG) >= vanaDay() then
+        player:messageText(npc, zones[zoneID].text.EGG_HUNT_OFFSET + messageOffset.HINTING)
+    else
+        local party = player:getParty()
+        local options = -1
+
+        for i = 1, #party do
+            if
+                party[i]           ~= nil and                -- Player exists
+                player:getName()   ~= party[i]:getName() and -- Different player
+                player:getZoneID() == party[i]:getZoneID()   -- Same zone
+            then
+                local initial = string.sub(party[i]:getName(), 1, 1)
+                local letter = string.byte(string.lower(initial)) - 97
+                options = utils.mask.setBit(options, letter, 0)
+            end
+        end
+
+        if options ~= -1 then
+            options = utils.mask.setBit(options, 26, 0)
+            player:startEvent(xi.events.egg_hunt.data[zoneID].cs, 3, options)
+        else
+            player:startEvent(xi.events.egg_hunt.data[zoneID].cs, 1)
+        end
+    end
+end
+
+xi.events.egg_hunt.onEventFinish = function(player, csid, option)
+    local zoneID = player:getZoneID()
+    local ID = zones[zoneID]
+
+    -- "Forget it." or out of range
+    if option > 28 then
```

Comment:
For bonus points here as well, this seems suspicious, and smells like a bitfield.  If >= 3 (0b11), I'd be willing to bet if you shift right two, its most likely a mask based on it from there.